packages/flow-core/src/components/f-toast/f-toast.ts

```diff
@@ -0,0 +1,289 @@
+	 * @returns uid
+	 */
+	generateId(length = 5) {
+		let result = "";
+		const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+		const charactersLength = characters.length;
+		for (let i = 0; i < length; i++) {
+			result += characters.charAt(Math.floor(Math.random() * charactersLength));
+		}
+		return result;
+	}
+
+	/**
+	 * remove the toast on click of remove icon
+	 */
+	remove() {
+		toastQueue.remove(this, () => {
+			this.dispatchOnRemove();
+		});
+	}
+
+	/**
+	 * checks for whether or not the toast auto-hides or persists
+	 * @returns void
+	 */
+	autoRemoveConfig() {
+		if (this.type === "persists") {
+			return;
+		}
+		setTimeout(() => {
+			if (!this.mouseover) {
+				this.remove();
+			} else {
+				this.autoRemoveConfig();
+			}
+		}, this.duration);
+	}
+
+	/**
+	 * set position of resepective toast
+	 * @param top top position string in px
+	 * @param right rigjt position string in px
+	 */
+	setPosition(top: string, right: string) {
+		this.top = top;
+		this.right = right;
+		this.requestUpdate();
+	}
+
+	/**
+	 * dispatch remove event
+	 * @param e Event
+	 */
+	dispatchOnRemove() {
+		/**
+		 * @event remove
+		 */
+		const event = new CustomEvent("remove", {
+			detail: {
+				value: this.uid
+			},
+			bubbles: true,
+			composed: true
+		});
+		this.dispatchEvent(event);
+	}
+
+	/**
+	 * display toast message f-div
+	 */
+	displayToastMessage() {
+		if (!this.hasToastMessage) {
+			this.messageSlot.style.display = "none";
+		} else {
+			this.messageSlot.style.display = "";
+		}
+	}
+
+	/**
+	 * display toast action f-div
+	 */
+	displayToastActions() {
+		if (!this.hasToastActions) {
+			this.actionsSlot.style.display = "none";
+		} else {
+			this.actionsSlot.style.display = "";
+		}
+	}
+
+	/**
+	 * add toast to queue
+	 */
+	addToastToQueue() {
+		if (this.toasterRef.value) {
+			this.height = this.toasterRef.value?.offsetHeight;
+			this.setAttribute("uid", this.generateId());
+			toastQueue.add(this);
+			this.autoRemoveConfig();
+		}
+	}
+
+	/**
+	 * Final render
+	 * @returns html
+	 */
+	render() {
+		// render empty string, since there no need of any child element
+		return html` <div
+			${ref(this.toasterRef)}
+			class="f-toast"
+			style=${this.styleObj}
+			data-position="fixed"
+			@mouseover=${() => {
+				this.mouseover = true;
+			}}
+			@mouseleave=${() => {
+				this.mouseover = false;
+			}}
+		>
+			${this["close-button"]
+				? html` <div class="f-toast-close" @click=${this.remove}>
+						<f-icon source="i-close" size="x-small" clickable></f-icon>
+				  </div>`
+				: ""}
+			<f-div
+				direction="column"
+				gap="medium"
+				state="tertiary"
+				width="100%"
+				padding="large none"
+				variant="curved"
+			>
+				<f-div
+					.selected=${this.state !== "default" ? "notch-left" : "none"}
+					padding="none large"
+					id="message-slot"
+					.state=${this.state}
+					toast-data-state=${this.state}
+				>
+					<slot name="message"></slot>
```

Comment:
please remove the message and action slot, since it is not adding any particular purpose.

To add a border for the state you can use pseudo-element